runtime/lua/vim/_editor.lua

```diff
@@ -876,6 +876,15 @@ function vim._init_default_mappings()
   -- Use : instead of <Cmd> so that ranges are supported. #19365
   map('n', '&', ':&&<CR>')
 
+  -- Repeat macro for each line of visual selection #21422
+  map('x', 'Q', ':normal @<c-r>=reg_recorded()<CR><CR>')
+  vim.api.nvim_set_keymap(
+    'x',
+    '@',
```

Comment:
For a default/builtin mapping it's not ideal to map "@" when we really mean "@x". I guess it's a start, though.